core/src/com/unciv/models/ruleset/unique/Conditionals.kt

```diff
@@ -78,6 +83,42 @@ object Conditionals {
             return compare(relevantCiv!!.getEraNumber(), era.eraNumber)
         }
 
+        /** Helper for comparing the Resource and Stat numbers **/
+        fun compareResourceOrStatAmount(compare: (first: Int, second: Int) -> Boolean): Boolean {
+            if (gameInfo == null) return false
+            val first = condition.params[0]
+            val second = condition.params[1]
+
+            // Compare Resource with Resource
+            if (checkOnResourceName(first) && checkOnResourceName(second)) {
```

Comment:
Entirely wedding way to go around this.
You should check each element indicidually to get number A and B.
Then compare.

Reply:
Done. Check it out.

Reply:
> Done. Check it out.

Below are my suggestion to optimize the code, not tested yet, however

```kotlin
        fun getStatOrResourceAmount(param: String): Int? {
            return if (gameInfo!!.ruleset.tileResources.containsKey(param)) getResourceAmount(param)
            else if (Stat.isStat(param)) {
                if (relevantCiv != null)
                    relevantCity?.getStatReserve(Stat.safeValueOf(param)!!)
                relevantCiv?.getStatReserve(Stat.safeValueOf(param)!!)
            }
            else null
        }

        /** Helper for comparing the Resource and Stat numbers **/
        fun compareResourceOrStatAmount(compare: (first: Int, second: Int) -> Boolean): Boolean {
            if (gameInfo == null) return false
            return compare(
                getStatOrResourceAmount(condition.params[0]) ?: return false,
                getStatOrResourceAmount(condition.params[1]) ?: return false
            )
        }
```